src/Middleware/OutputCaching/src/OutputCacheMiddleware.cs

```diff
@@ -200,6 +200,7 @@ private async Task InvokeAwaited(HttpContext httpContext, IReadOnlyList<IOutputC
         {
             // avoid recycling in unknown outcomes, especially re concurrent buffer access thru cancellation
             hasException = true;
+            throw;
```

Comment:
Should add a regression test

Reply:
yeah, I was planning to - [have an initial test](https://github.com/dotnet/aspnetcore/pull/51067/commits/135c4fa9fd9688c57840b78a3fb713ab8e22cf19), but it manifests contrary to my expectations (i.e. we see an actual exception rather than a 5xx http); I'm going to check this test against the "before I touched anything" state, to see if this is just an artefact of the test harness

Reply:
tested at `b362c450058d7fe3279ffdcc1f6e40beaf828967` (which is "before") - behaviour is identical there, i.e. yes, I think that's a feature of the test harness (presumably we're hooking at a level that doesn't include the global fault loop)